### PR TITLE
QPPCT-618: Improved invalid program name error

### PR DESCRIPTION
**Pull request by halprin**
### Information
- JIRA story QPPCT-618.

### Changes proposed in this PR:
- Improves the error message when an invalid program name is supplied.
- The invalid program name error is no longer generated when the program name is completely missing.
- Updated `ERROR_MESSAGES.md`.

### Checklist 
- [x] All JUnit tests pass (`mvn clean verify`).
- [x] New unit tests written to cover new functionality.
- [x] Added and updated JavaDocs for non-test classes and methods.
- [x] No local design debt. Do you feel that something is "ugly" after your changes?
- [x] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: commons/src/main/java/gov/cms/qpp/conversion/model/error/ErrorCode.java [hunk 1]
```diff
@@ -57,7 +57,8 @@
 	CLINICAL_DOCUMENT_MISSING_ACI_OR_IA_OR_ECQM_CHILD(23, "Clinical Document Node must have at least one "
 			+ "Aci or IA or eCQM Section Node as a child"),
 	CLINICAL_DOCUMENT_MISSING_PROGRAM_NAME(24, "Clinical Document must have one and only one program name"),
-	CLINICAL_DOCUMENT_INCORRECT_PROGRAM_NAME(25, "Clinical Document program name is not recognized"),
+	CLINICAL_DOCUMENT_INCORRECT_PROGRAM_NAME(25, "The Clinical Document program name `(program name)` is not recognized. Valid "
+		+ "program names are MIPS_GROUP, MIPS_INDIV, or CPCPLUS.", true),
```
Comment (halprin) [inline commons/src/main/java/gov/cms/qpp/conversion/model/error/ErrorCode.java]: If mulling over how to programmatically fill in the valid program names by using the `Program` enum.

File: converter/src/main/java/gov/cms/qpp/conversion/model/Program.java [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 import java.util.stream.Collectors;
 
-import gov.cms.qpp.conversion.decode.ClinicalDocumentDecoder;
-
 /**
  * Construct that helps categorize submissions by program name.
  */
 public enum Program {
-	MIPS("MIPS_GROUP", "MIPS_INDIV", "MIPS"),
+	MIPS("MIPS_GROUP", "MIPS_INDIV"),
```
Comment (saquino0827) [inline converter/src/main/java/gov/cms/qpp/conversion/model/Program.java]: Do we not accept 'MIPS' anymore?

Comment (halprin) [reply converter/src/main/java/gov/cms/qpp/conversion/model/Program.java]: Correct.  But that should be OK.

File: commons/src/main/java/gov/cms/qpp/conversion/util/StringHelper.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	/**
+	 * Joins a {@link Iterable} of {@link String}s together with the specified separator. Between the second to last and last
+	 * item, the specified conjunction is used instead.  This allows one to make a {@link String} like {@code Dog, Cow, and Moof}.
+	 *
+	 * @param iterable The object to join together
+	 * @param separator The {@link String} that splits most of the items in the iterable.
+	 * @param conjunction The {@link String} that splits the second to last and last item.
+	 * @return A joined {@link String}.
+	 */
+	public static String join(Iterable<String> iterable, String separator, String conjunction) {
```
Comment (omnivion) [inline commons/src/main/java/gov/cms/qpp/conversion/util/StringHelper.java]: what's wrong with Collectors#joining (think String#join is a thing too?)

Comment (halprin) [reply commons/src/main/java/gov/cms/qpp/conversion/util/StringHelper.java]: None of them allow for a custom conjunction between the second-to-last and last element.  Sadly, there isn't a clean way to build these frameworks into `StringHelper#join` either without being inefficient or not allowing a generic `Collection` or `Iterable`.

**Pull request merged**